apps/system/dat/dat_main.c

```diff
@@ -0,0 +1,337 @@
+	if (ret	< 0) {
+		printf("ERROR executing MPU ioctl ERROR = %d\n", errno);
+	}
+
+	printf("Number of MPU regions: %d\n\n", ret);
+	printf("End of MPU Information\n");
+
+	close(tc_fd);
+}
+#endif
+
+/****************************************************************************
+ * Name:  dat_tcb_info
+ *
+ * Description:
+ *   This function displays the TCB information of all the tasks
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+void dat_tcb_info(void)
+{
+	int ret = 0;
+	int tc_fd = 0;
+
+	/* Get number of binaries */
+
+	tc_fd = open(OS_API_TEST_DRVPATH, O_WRONLY);
+	if (tc_fd < 0) {
+		printf("FAILED to open OS API test driver %s\n", OS_API_TEST_DRVPATH);
+	}
+
+	printf("TCB Information >>\n");
+
+	ret = ioctl(tc_fd, TESTIOC_GET_TCB_INFO, 0);
+	if (ret	< 0) {
+		printf("ERROR executing TCB ioctl ERROR = %d\n", errno);
+	}
+
+	ret = ioctl(tc_fd, TESTIOC_GET_TCBLIST_INFO, 0);
+	if (ret	< 0) {
+		printf("ERROR executing TCB ioctl ERROR = %d\n", errno);
+	}
+
+	ret = ioctl(tc_fd, TESTIOC_GET_TCBSTACK_INFO, 0);
+	if (ret	< 0) {
+		printf("ERROR executing TCB ioctl ERROR = %d\n", errno);
+	}
+
+	printf("End of TCB Information\n");
+
+	close(tc_fd);
+}
+
+/****************************************************************************
+ * Name:  dat_app_info
+ *
+ * Description:
+ *   This function displays the APPLICATION information
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+void dat_app_info(void)
+{
+	int ret = 0;
+	int tc_fd = 0;
+
+	printf("Application Information >>\n");
+
+	/* Get number of binaries */
+
+	tc_fd = open(OS_API_TEST_DRVPATH, O_WRONLY);
+	if (tc_fd < 0) {
+		printf("FAILED to open OS API test driver %s\n", OS_API_TEST_DRVPATH);
+		return;
+	}
+
+	ret = ioctl(tc_fd, TESTIOC_GET_APP_COUNT, 0);
+	if (ret	< 0) {
+		printf("ERROR executing Application ioctl ERROR = %d\n", errno);
+		return;
+	}
+
+	printf("Number of binaries : %d\n\n", ret);
+
+	/* Calculate app running time */
+
+	clock_t ticktime;
+#ifdef CONFIG_SYSTIME_TIME64
+	uint64_t sec;
+	uint64_t start_sec;
+#else
+	uint32_t sec;
+	uint32_t start_sec;
+#endif
+	unsigned int csec;
+	unsigned int start_csec;
+
+	ticktime = clock();
+
+	/* Convert the system up time to seconds + hundredths of second */
+	sec = ticktime / CLOCKS_PER_SEC;
+	csec = (100 * (ticktime % CLOCKS_PER_SEC) + (CLOCKS_PER_SEC / 2)) / CLOCKS_PER_SEC;
+
+	/* Make sure that rounding did not force the hundredths of a second above 99 */
+	if (csec > 99) {
+		sec++;
+		csec -= 100;
+	}
+
+	/* Retrieve app1 start timings */
+
+	start_sec = ioctl(tc_fd, TESTIOC_GET_APP_START_TIME_SEC, 1);
+	if (start_sec < 0) {
+		printf("ERROR executing Application ioctl ERROR = %d\n", errno);
```

Comment:
Could you modify all such debug lines a little?
"ERROR ....... ERROR"
Could you add a comma/restructure the debug statement please?